src/backend/dx11/src/lib.rs

```diff
@@ -1312,6 +1316,25 @@ impl hal::command::RawCommandBuffer<Backend> for CommandBuffer {
     }
 }
 
+bitflags! {
+    struct MemoryHeapType: u64 {
+        const DEVICE_LOCAL = 1 << 0;
+        const HOST_NONCOHERENT = 1 << 1;
+        const HOST_COHERENT = 1 << 2;
+    }
+}
+
+impl MemoryHeapType {
+    pub fn from_raw(index: usize) -> Option<Self> {
+        match Self::from_bits(1 << index).unwrap() {
+            Self::DEVICE_LOCAL => Some(Self::DEVICE_LOCAL),
+            Self::HOST_NONCOHERENT => Some(Self::HOST_NONCOHERENT),
+            Self::HOST_COHERENT => Some(Self::HOST_COHERENT),
+            _ => None
```

Comment:
how is this possible?

Reply:
Invalid input? I suppose that's not really my job though :p. Also noticed there is a `from_bits_truncate` in bitflags which is more suitable.